src/Datastore/EntityMapper.php

```diff
@@ -429,4 +446,18 @@ private function isEncoded($value)
 
         return true;
     }
+
+    /**
+     * Get the value type from a value object.
+     *
+     * @param array $value
+     * @return string
+     */
+    private function getValueType(array $value)
+    {
+        $keys = array_keys($value);
+        return array_values(array_filter($keys, function ($key) {
+            return in_array($key, $this->valueTypes);
+        }))[0];
```

Comment:
I updated this to a `RuntimeException`. It is not possible to save a property without a value, so the only time this exception should be hit (outside of tests etc) would be if there were a malformed entity (not sure if this is possible) or an error in the service.